cmd/fixenv.go

```diff
@@ -0,0 +1,700 @@
+	KeyToPrincipalKey map[string]string
+	Operators         map[string]*OT
+}
+
+type OT struct {
+	OC         jwt.OperatorClaims
+	Jwts       map[string]string
+	Accounts   map[string]*jwt.AccountClaims
+	ActToUsers map[string]jwt.StringList
+}
+
+func NewOT() *OT {
+	var ot OT
+	ot.Jwts = make(map[string]string)
+	ot.Accounts = make(map[string]*jwt.AccountClaims)
+	ot.ActToUsers = make(map[string]jwt.StringList)
+	return &ot
+}
+
+func createFixCmd() *cobra.Command {
+	var params FixCmd
+
+	var cmd = &cobra.Command{
+		Use:   "fix",
+		Short: "rebuilds a project tree from screwed the contents of one or more directories",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := RunStoreLessAction(cmd, args, &params); err != nil {
+				return err
+			}
+			if params.operators > 0 {
+				cmd.Printf("stored %d operators\n", params.operators)
+				cmd.Printf("stored %d accounts\n", params.accounts)
+				cmd.Printf("stored %d users\n", params.users)
+				cmd.Printf("stored %d nkeys\n", params.nkeys)
+				cmd.Println()
+				if params.nkeys > 0 {
+					cmd.Printf("export NKEYS_PATH=%s\n", filepath.Join(params.out, "keys"))
+					cmd.Printf("cd %s\n", filepath.Join(params.out, "operators"))
+				} else {
+					cmd.Printf("cd %s\n", filepath.Join(params.out, "operators"))
+				}
+				cmd.Println("nsc list operators")
+			}
+			return nil
+		},
+		Hidden: true,
+	}
+
+	cmd.Flags().StringSliceVarP(&params.in, "in", "", nil, "input paths")
+	cmd.Flags().StringVarP(&params.out, "out", "", "", "output dir")
+	cmd.Flags().BoolVarP(&params.creds, "creds", "", false, "import creds")
+	cmd.MarkFlagRequired("in")
+
+	return cmd
+}
+
+func (p *FixCmd) SetDefaults(ctx ActionCtx) error {
+	p.Keys = make(map[string]string)
+	p.KeyToPrincipalKey = make(map[string]string)
+	p.Operators = make(map[string]*OT)
+	return nil
+}
+
+func (p *FixCmd) PreInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *FixCmd) Load(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *FixCmd) PostInteractive(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *FixCmd) Validate(ctx ActionCtx) error {
+	return nil
+}
+
+func (p *FixCmd) Run(ctx ActionCtx) (store.Status, error) {
+	ctx.CurrentCmd().SilenceUsage = true
+	rr := store.NewReport(store.OK, "")
+	// Load all the keys
+	kr, err := p.LoadNKeys(ctx)
+	rr.Add(kr)
+	if err != nil {
+		return rr, err
+	}
+
+	lr, err := p.LoadOperators(ctx)
+	rr.Add(lr)
+	if err != nil {
+		return rr, err
+	}
+
+	ar, err := p.LoadAccounts(ctx)
+	rr.Add(ar)
+	if err != nil {
+		return rr, err
+	}
+
+	ur, err := p.LoadUsers(ctx)
+	rr.Add(ur)
+	if err != nil {
+		return rr, err
+	}
+
+	if p.creds {
+		cr, err := p.LoadCreds(ctx)
+		rr.Add(cr)
+		if err != nil {
+			return rr, err
+		}
+	}
+
+	if len(p.Operators) == 0 {
+		return nil, fmt.Errorf("no operator found")
+	}
+
+	if err := p.Regenerate(rr); err != nil {
+		return rr, err
+	}
+
+	return rr, nil
+}
+
+func (p *FixCmd) Regenerate(rr *store.Report) error {
+	if p.out == "" {
```

Comment:
So this will create a "fix_" directory.. Should it be deleted in case of failure? And on success, should that be renamed? How user would use this "fix"?

Reply:
The tool will actually print what you have to do, I also added an example section - note the tool is hidden and won't show for anyone unless they know it is there.

The only place where it can fail is on reading files, and creating the output directories. If it fails, you should be able to repeat the imports by adding new directories along with the 'fixed' one

Things that will fail are:
- no operator is found
- account references an unknown operator
- user references an unknown account

There's also an option to load cred files, which will read the cred files and import jwts and nkeys from them.